pkg/kbld/config/config.go

```diff
@@ -167,12 +175,15 @@ func NewConfigFromImagesLock(res ctlres.Resource) (Config, error) {
 	overridesConfig := NewConfig()
 
 	for _, image := range imagesLock.Images {
+		var iMetas []ImageMeta
+		yaml.Unmarshal([]byte(image.Annotations[ImagesLockKbldMetas]), &iMetas)
```

Comment:
If an imgpkg lock file cannot be marshalled into an ImageMeta struct, we should report an error that the lock file format was incorrect.